js/index.js

```diff
@@ -1 +1,107 @@
 // Your code goes here
+
+// 1. mouseover
+const imgContent = Array.from(document.querySelectorAll('.img-content'));
+
+imgContent.forEach(img => {
+  img.addEventListener('mouseover', () => {
+    TweenMax.to(img.firstElementChild, 1, {
+      height: '99%',
+      opacity: 1,
+      ease: Power4.easeInOut
+    });
+  });
+});
+
+// 2. mouseout
+imgContent.forEach(img => {
+  img.firstElementChild.addEventListener('mouseleave', () => {
+    TweenMax.to(img.firstElementChild, 0.5, {
+      height: 0,
+      opacity: 0,
+      ease: Power4.easeInOut
+    });
+  });
+});
+
+// 3. doubleclick
+const hoverBtns = document.querySelectorAll('.btn-hover');
+const adventureImgs = document.querySelectorAll('.adventure-img');
+
+hoverBtns.forEach((btn, i) => {
+  btn.addEventListener('dblclick', () => {
+    adventureImgs[i].src = 'https://picsum.photos/384/288';
+  });
+});
+
+// 4. keyup
+const filterInput = document.querySelector('#filterInput');
+const container = document.querySelector('.container');
+const logoHeading = document.querySelector('.logo-heading');
+
+filterInput.addEventListener('keyup', () => {
+  // change logo text
+  logoHeading.innerText = filterInput.value;
```

Comment:
pretty cool understanding of the input, this sort of thing gets really cool in React with "controlled components"